pkg/reconcilermanager/controllers/util.go

```diff
@@ -79,43 +88,61 @@ func hydrationEnvs(sourceType string, gitConfig *v1beta1.Git, ociConfig *v1beta1
 		// Add Hydration Polling Period.
 		corev1.EnvVar{
 			Name:  reconcilermanager.HydrationPollingPeriod,
-			Value: pollPeriod,
+			Value: opts.pollPeriod,
 		})
 	return result
 }
 
+type reconcilerOptions struct {
+	clusterName       string
+	syncName          string
+	syncGeneration    int64
+	reconcilerName    string
+	reconcilerScope   declared.Scope
+	sourceType        string
+	gitConfig         *v1beta1.Git
```

Comment:
nit: Do we want to pass in pointers here? Granted there are some tradeoffs, but since we have no intention of modifying the variable in this function I feel it may be clearer to pass by value. WDYT?

Reply:
Ideally we should pass by value, but there is some trick here.  The spec.git field is a pointer. If we pass by value, we need to check for nil first. If it is nil, then build an empty struct. Only one of the spec is not nil, so we need to build empty struct for the other two.
We can make it better, but seems another refactor.

Reply:
Ack, SGTM